File: src/main/java/com/smart/jtesting/model/Item.java
package com.smart.jtesting.model;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;
import java.io.Serializable;

@Entity
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Table(name = "Item")
public class Item implements Serializable {

    @Id
    @Column(name = "id", nullable = false, unique = true)
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @Column(name = "code", nullable = false, unique = true)
    private String code;

    @Column(name = "name")
    private String name;

    @Column(name = "price")
    private Double price;

    @Column(name = "category")
    @Enumerated(EnumType.STRING)
    private ItemCategory category;

    public Item(String code, String name, Double price, ItemCategory category) {
        this.code = code;
        this.name = name;
        this.price = price;
        this.category = category;
    }
}

File: src/main/webapp/resources/JTesting/app/model/Item.js
Ext.define('JTesting.model.Item', {
    extend: 'Ext.data.Model',        
    fields: [
        {name: 'id', type: 'int'},
        {name: 'code', type: 'string'},
        {name: 'name', type: 'string'},
        {name: 'price', type: 'double'},
        {name: 'category', type: 'string'}
    ]
});